pkgs/shells/powershell/default.nix

```diff
@@ -0,0 +1,46 @@
+{ stdenv, fetchgit, autoPatchelfHook, fetchzip, libunwind, libuuid, icu, curl, cacert,
+  makeWrapper, less, openssl }:
+
+let platformString = if stdenv.isDarwin then "osx"
+                     else if stdenv.isLinux then "linux"
+                     else throw "unsupported platform";
+    platformSha = if stdenv.isDarwin then "1ga4p8xmrxa54v2s6i0q1q7lx2idcmp1jwm0g4jxr54fyn5ay3lf"
+                     else if stdenv.isLinux then "000mmv5iblnmwydfdvg5izli3vpb6l14xy4qy3smcikpf0h87fhl"
+                     else throw "unsupported platform";
+    platformLdLibraryPath = if stdenv.isDarwin then "DYLD_FALLBACK_LIBRARY_PATH"
+                     else if stdenv.isLinux then "LD_LIBRARY_PATH"
+                     else throw "unsupported platform";
+in
+stdenv.mkDerivation rec {
+  name = "powershell-${version}";
+  version = "6.0.2";
+
+  src = fetchzip {
+    url = "https://github.com/PowerShell/PowerShell/releases/download/v${version}/powershell-${version}-${platformString}-x64.tar.gz";
+    sha256 = platformSha;
+    stripRoot = false;
+  };
+
+  buildInputs = [ autoPatchelfHook makeWrapper ];
+  propagatedBuildInputs = [ libunwind libuuid icu curl cacert less openssl ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/share/powershell
+    cp -r * $out/share/powershell
+    rm $out/share/powershell/DELETE_ME_TO_DISABLE_CONSOLEHOST_TELEMETRY
+    makeWrapper $out/share/powershell/pwsh $out/bin/pwsh --prefix ${platformLdLibraryPath} : "${stdenv.lib.makeLibraryPath [ libunwind libuuid icu openssl curl ]}" \
+                                           --set PAGER ${less}/bin/less --set TERM linux
```

Comment:
Sorry for the incremental review, but I just noticed that TERM is that for linux. It currently does not like screen as a terminal, however if we add workaround should it not better be `xterm`, because I suppose most people are using a graphical terminal that can be resized rather a virtual console.

Is less as pager strictly required? An alternative could be to only set less if no other PAGER is specified.

Reply:
The reason for this PAGER workaround is that if I omit this, this is what I'll get:

```

PS /home/yrashk/Projects/nix/nixpkgs> help                                                                                                                                                                          
Get-Command : The term 'less -R' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```

Any other ideas on how to work around this?

I've changed the terminal to be xterm, it works fine.

Reply:
Should be PAGER not rather an executable then a command with arguments? There is the LESS variable to pass arguments to PAGER.

Reply:
@Mic92 I'm not sure what you mean exactly

Reply:
As for the pager issue, it appears to be fixed in a preview version of PowerShell (v6.1.0-preview.1 - 2018-03-23) as per their changelog (search for 6144) https://github.com/PowerShell/PowerShell/pull/6144

I suppose we have two options:

* Use a preview build instead of the release
* Keep PAGER set as it is right now but add a TODO: "remove after upgrading to v6.1.0-preview.1 or later" or similar.

Suggestions?

Reply:
Ok. Please add the TODO then.

Reply:
@Mic92 added TODO, pushed

Anything else?